### PR TITLE
Allow creating API keys with an expiration date

### PR DESCRIPTION
**Pull request by segiddins**
Will be used for CLI workflow to add trusted publishing to a gem, since we dont want the resulting API key that can manage trusted publishing to be long-lived

### CODE REVIEW

File: config/locales/de.yml [hunk 1]
```diff
@@ -7,19 +7,20 @@ de:
   verification_expired:
   feed_latest: RubyGems.org | Neueste Gems
   feed_subscribed: RubyGems.org | Abonnierte Gems
-  footer_about_html:
```
Comment (martinemde) [inline config/locales/de.yml:10]: Since bin/fill_locales wants to make a bunch of changes, could you submit a separate pr with just that?

File: test/system/api_keys_test.rb [hunk 2]
```diff
@@ -153,6 +153,21 @@ class ApiKeysTest < ApplicationSystemTestCase
     assert_predicate @user.api_keys.last, :mfa_enabled?
   end
 
+  test "creating new api key with expiration" do
```
Comment (martinemde) [inline test/system/api_keys_test.rb:156]: Can you update expiration on a key? Maybe a test for that.

**Review by martinemde — COMMENTED** [review]
Looking good so far.

File: test/system/api_keys_test.rb (hunk 2 above)
Comment (indirect) [reply test/system/api_keys_test.rb:156]: IMO we don't want to allow updating expiration after the fact, and you should generate a new key if you need a new expiration.

Comment (segiddins) [reply test/system/api_keys_test.rb:156]: It's disabled in the UI and also in the controller, will add a functional test for the latter

**Review by martinemde — APPROVED** [review]

File: app/views/api_keys/_form.html.erb [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <%= form_with model: [:profile, @api_key], data: { controller: "exclusive-checkbox gem-scope" } do |f| %>
   <%= label_tag :name, t("api_keys.index.name"), class: "form__label" %>
-  <%= f.text_field :name, class: "form__input", autocomplete: :off %>
+  <%= f.text_field :name, class: "form__input", autocomplete: :off, disabled: f.object.persisted? %>
```
Comment (simi) [inline app/views/api_keys/_form.html.erb:3]: Is name "readonly" once created? Is that enforced also on server side somehow?

Comment (segiddins) [reply app/views/api_keys/_form.html.erb:3]: yes, enforced in the controllers and there are tests for that

Comment (simi) [reply app/views/api_keys/_form.html.erb:3]: Looking again, it seems covered by controller. Is that enough? What about model validation?
